src/components/PostDetails.tsx

```diff
@@ -124,33 +73,45 @@ export const PostDetails: React.FC<Props> = ({ post }) => {
           <>
             <p className="title is-4">Comments:</p>
 
-            {comments.map(comment => (
-              <article
-                className="message is-small"
-                key={comment.id}
-                data-cy="Comment"
-              >
-                <div className="message-header">
-                  <a href={`mailto:${comment.email}`} data-cy="CommentAuthor">
-                    {comment.name}
-                  </a>
-
-                  <button
-                    data-cy="CommentDelete"
-                    type="button"
-                    className="delete is-small"
-                    aria-label="delete"
-                    onClick={() => deleteComment(comment.id)}
-                  >
-                    delete button
-                  </button>
-                </div>
-
-                <div className="message-body" data-cy="CommentBody">
-                  {comment.body}
-                </div>
-              </article>
-            ))}
+            {comments.map(comment => {
+              const {
+                id,
+                email,
+                name,
+                body,
+              } = comment;
+
+              return (
+                <article
+                  className="message is-small"
+                  key={id}
+                  data-cy="Comment"
+                >
+                  <div className="message-header">
+                    <a href={`mailto:${email}`} data-cy="CommentAuthor">
+                      {name}
+                    </a>
+
+                    <button
+                      data-cy="CommentDelete"
+                      type="button"
+                      className="delete is-small"
+                      aria-label="delete"
+                      onClick={() => {
+                        dispatch(remove(id));
+                        dispatch(removeAsyncComment(id));
```

Comment:
Pls move the function inside `onClick` handler to a separate const and use as a callback here

Reply:
Not fixed. 
![image](https://github.com/mate-academy/react_redux-list-of-posts/assets/99403198/5e50f18d-6c09-4de5-8713-a737b306943f)
Move the onClick logic to a separate function. It follows the best practice of separating concerns and keeping your JSX code clean and focused on rendering components rather than defining complex logic directly within it.